### ISSUE TITLE
Please update latest version to 5.8.0

### ISSUE THREAD

**Issue by twenzel** [NuGet/NuGetGallery#8319]
Please update the `https://dist.nuget.org/win-x86-commandline/latest/nuget.exe` version to 5.8.0 instead of 5.7.0.

There's a problem when using NuGet Restore with 5.7 and MSBuild 16.8 (https://developercommunity.visualstudio.com/content/problem/1248649/error-netsdk1005-assets-file-projectassetsjson-doe.html)

Changing the "latest" version to 5.8.0 would fix many installations using cake-build (Or has @patriksvensson a better solution?).

**Comment by loic-sharma** [comment]
Hello, thank you for opening this issue! We are currently blocked on updating the latest version of nuget.exe until Azure DevOps has updated their build agents for their hosted services.

Could you provide explain how and why you are using `https://dist.nuget.org/win-x86-commandline/latest/nuget.exe`?  For your information, there is a protocol available to discover nuget.exe versions: https://docs.microsoft.com/en-us/nuget/api/tools-json

Ideally, you should use the version of nuget.exe that corresponds to your build tools. If possible, you should prefer using MSBuild to restore instead of nuget.exe.

**Comment by twenzel** [comment]
The cake-build bootstrapper (https://cakebuild.net/docs/running-builds/runners/cake-runner-for-dotnet-framework#bootstrapping-for-cake-runner-for.net-framework) is using this url to download the "latest" version of nuget.exe to retrieve its modules nuget packages for the "plugins".
After installing the latest MSBuild Version (16.8) on our build agents our builds crashes. 
The problem is that in our build logic we're executing a dedicated `nuget restore` on our solution. There, the downloaded NuGet.exe (5.7) is used. After the `restore` MSBuild is executed, using the installed 16.8 version, which requires a `project.asset.json` in a newer format that nuget.exe 5.7 has created.

As workaround for the development branch (and all coming versions) we're now using MSBuild to restore the solution instead of a dedicated nuget.exe restore (as you already suggested). 
But there might be scenarios where older branches cannot be build anymore.

**Comment by loic-sharma** [comment]
I'm glad you were able to find a workaround. We've published a blog post for nuget 5.8 that further explains this problem and potential solutions: https://devblogs.microsoft.com/nuget/getting-started-with-nuget-5-8/#known-issues

I'll close this issue off for now. Feel free to create a new issue if you run into any additional problems!
